lint/capability_fields_analyser.go

```diff
@@ -74,6 +107,32 @@ var CapabilityFieldAnalyzer = (func() *analysis.Analyzer {
 					if !ok {
 						return
 					}
+
+					nt, ok := field.TypeAnnotation.Type.(*ast.NominalType)
+					if ok {
+						hasCapability, found := compositeIdsCapabilitiesPropery[nt.Identifier.Identifier]
+
+						if found {
+							if hasCapability && field.Access == ast.AccessPublic {
```

Comment:
same here:
`if !hasCapability { return }`
`if field.Access != ast.AccessPublic { return }
// report here 
` ?

Reply:
Please refer to this comment https://github.com/onflow/cadence-tools/pull/176#discussion_r1272119853